### PR TITLE
Fixed the `keygenerator` binary

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
the keyGenerator binary produced malformed p2pKey.pem files like:
```
-----BEGIN PRIVATE KEY for ����Q6~�1M*��c�I�D��HV ��k���q�-----
YjgwOGM1NTFmMWM5MTM4ODhiZmRhOTIyMmU0MTE1ZjRlYTM1M2Y2NDQ2MGFlMzZk
ZjM5MTkwZDA5NThhZTM4MA==
-----END PRIVATE KEY for ����Q6~�1M*��c�I�D��HV ��k���q�-----
```
  
## Proposed changes
- fixed keyGenerator binary for p2p key-type
- integrated new elrond-go-p2p repo version

## Testing procedure
- standard system test
- generate a new p2pKey.pem file by using the the following command: `keygenerator -key-type p2p`. The contents should have the correct pem file generated like:
```
-----BEGIN PRIVATE KEY for 16Uiu2HAm3akMrz4uTE85nDdSac7XL9uJYxS8iDUHwSywgB3ruC4K-----
NDI2ZThlYWM2ZGUyYzI0MTc1NDlmYTVhNTE0YThkNGQ1YjAwMjU2N2NhZGMwNThk
Y2YyNTY2MTJhNThmYmZkZA==
-----END PRIVATE KEY for 16Uiu2HAm3akMrz4uTE85nDdSac7XL9uJYxS8iDUHwSywgB3ruC4K-----
```

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/keygenerator/main.go [hunk 1]
```diff
@@ -407,6 +407,8 @@ func saveSliceKeys(baseFilenameTemplate string, keys []key, converter pubKeyConv
 	var file *os.File
 	var err error
 	for i, k := range keys {
+		log.Warn(hex.EncodeToString(k.pkBytes))
+
```
Comment (ssd04) [inline cmd/keygenerator/main.go:411]: do we want to keep this?

File: cmd/keygenerator/main.go [hunk 2]
```diff
@@ -407,6 +407,8 @@ func saveSliceKeys(baseFilenameTemplate string, keys []key, converter pubKeyConv
 	var file *os.File
 	var err error
 	for i, k := range keys {
+		log.Warn(hex.EncodeToString(k.pkBytes))
```
Comment (AdoAdoAdo) [inline cmd/keygenerator/main.go:410]: do we need this warn?

File: cmd/keygenerator/main.go (hunk 1 above)
Comment (iulianpascalau) [reply cmd/keygenerator/main.go:411]: no, left by mistake, will remove

**Review by AdoAdoAdo — APPROVED** [review]

File: cmd/keygenerator/main.go (hunk 2 above)
Comment (iulianpascalau) [reply cmd/keygenerator/main.go:410]: no, was used to generate a test data, will remove

**Review by AdoAdoAdo — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
system test passed

**Pull request merged**